src/Persistence/PersistentObjectFactory.php

```diff
@@ -275,8 +275,25 @@ protected function normalizeParameter(string $field, mixed $value): mixed
             $value->persist = $this->persist; // todo - breaks immutability
         }
 
-        if ($value instanceof self && Configuration::instance()->persistence()->relationshipMetadata(static::class(), $value::class(), $field)?->isCascadePersist) {
-            $value->persist = false;
+        if ($value instanceof self) {
+            $pm = Configuration::instance()->persistence();
+
+            $relationshipMetadata = $pm->relationshipMetadata($value::class(), static::class(), $field);
+
+            // handle inversed OneToOne
+            if ($relationshipMetadata && !$relationshipMetadata->isCollection && $inverseField = $relationshipMetadata->inverseField) {
+                $this->tempAfterPersist[] = static function(object $object) use ($value, $inverseField, $pm) {
+                    $value->create([$inverseField => $object]);
+                    $pm->refresh($object);
+                };
+
+                // creation delegated to afterPersist hook - return empty array here
+                return null;
```

Comment:
I'm wondering if it is OK to return null here.
Can a "OneToOne" relationship be "not nullable" on the both sides of the relationship?

Reply:
I think this very specific case, where both side of `OneToOne` are mandatory, could not be fixed : it is a circular dependency. Maybe we could find a way to emit a meaningful error for the user?

Also, maybe we could find a way to return something else than `null` here: in Foundry2, we used to return a `PostPersistCallback` which was not added to the normalized parameters. Maybe we could mimic this (but it would complexify the whole normalization phase)

Reply:
Yes, I don't remember the exact context of what happened but I think you're right, it's the circular dependency issue.